XenAdmin/Wizards/ImportWizard/ImportSelectNetworkPage.cs

```diff
@@ -51,6 +53,32 @@ protected override bool ImplementsIsDirty()
             return true;
         }
 
+        public override bool ShowNetwork(Host targetHost, XenAPI.Network network)
+        {
+            if (!network.Show(Properties.Settings.Default.ShowHiddenVMs))
+                return false;
+
+            if (network.IsSlave())
+                return false;
+
+            if (network.IsSriov() && !Allow_Sriov_Network())
+                return false;
+
+            if (targetHost != null && !targetHost.CanSeeNetwork(network))
+                return false;
+
+            if (targetHost == null && !network.AllHostsCanSeeNetwork())
+                return false;
+
+            return true;
+        }
+
+        private bool Allow_Sriov_Network()
+        {
+            string xml = OVF.ToXml(SelectedOvfEnvelope);
+            return xml.Contains("allow-network-sriov");

```

Comment:
If you export a VM into OVF format and open the ovf file in a text editor, you'll see that the `allow-network-sriov` is saved in the envelope with a value; here you're checking only the existence of the string; is the value of no importance?

Reply:
Also, we can export more than one VM in an OVF file, so in theory we can have some VMs with the allow-network-sriov flag present, others without it. This code will allow the selection of the SR-IOV network if one of the VMs has the flag on.

Reply:
To elaborate a bit more on Mihaela's comment, we should not be looking randomly in the xml for this string, but rather look in the specific other configuration data section of the machine. Also, there should be a method in the OVF class to do this - I know ToXml is public, but it should probably not be; we should not be handling the xml in the UI level, we should rather have a method in the OVF class to look for what we need.